src/app/ReadClient.cpp

```diff
@@ -134,85 +126,68 @@ CHIP_ERROR ReadClient::SendReadRequest(ReadPrepareParams & aReadPrepareParams)
     // TODO: SendRequest parameter is too long, need to have the structure to represent it
     CHIP_ERROR err = CHIP_NO_ERROR;
     System::PacketBufferHandle msgBuf;
-    ChipLogDetail(DataManagement, "%s: Client[%u] [%5.5s]", __func__,
-                  InteractionModelEngine::GetInstance()->GetReadClientArrayIndex(this), GetStateStr());
-    VerifyOrExit(ClientState::Initialized == mState, err = CHIP_ERROR_INCORRECT_STATE);
-    VerifyOrExit(mpCallback != nullptr, err = CHIP_ERROR_INCORRECT_STATE);
+    ChipLogDetail(DataManagement, "%s: Client[%lx] [%5.5s]", __func__, (uintptr_t) this, GetStateStr());
```

Comment:
TODO: Figure out right PRI incantation to use to print out pointers.

Reply:
"%p" instead of trying to print it as integer, no?